### PR TITLE
chore(client): Standardize on `[cC]hildView`, replace all `[sS]ub[vV]iew` instances.

### LINKED ISSUE #3173: Function and variable name inconsistency - sometimes SubView, sometimes Subview.
Pick one.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #3173

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
Slapping a WIP on this so nobody merges before the train is cut.

**Comment by shane-tomlinson** [comment]
No longer WIP, @philbooth, your it!

File: app/scripts/views/base.js [hunk 1]
```diff
@@ -431,33 +431,33 @@ function (Cocktail, _, Backbone, Raven, $, p, AuthErrors,
...
 
-      this.destroySubviews();
+      this.destroySubViews();
 
       this.trigger('destroyed');
     },
 
-    trackSubview: function (view) {
+    trackSubView: function (view) {
       if (! _.contains(this.subviews, view)) {
```
Comment (philbooth) [inline app/scripts/views/base.js]: Should we also update `subviews` to `subViews`?

File: app/tests/spec/views/base.js [hunk 2]
```diff
@@ -494,22 +494,22 @@ function (chai, $, sinon, BaseView, p, Translator, EphemeralMessages, Metrics,
       });
     });
 
-    describe('trackSubview/untrackSubview', function () {
-      it('trackSubview tracks a subview, untrackSubview untracks the subview', function () {
+    describe('trackSubView/untrackSubView', function () {
+      it('trackSubView tracks a subview, untrackSubView untracks the subview', function () {
         var subview = new BaseView();
```
Comment (philbooth) [inline app/tests/spec/views/base.js]: And, while I'm being all anal about stuff, this should be `subView`.

**Comment by philbooth** [comment]
This looks good to me, barring the two `subview` / `subviews` nits.

File: app/scripts/views/base.js (hunk 1 above)
Comment (shane-tomlinson) [inline app/scripts/views/base.js]: I dig it, I didn't think to search for that.

**Comment by shane-tomlinson** [comment]
Maybe in light of #3196, we should replace `[sS]ub[vV]iew` with `[cC]hildView]` in this PR and be done with it.

**Comment by shane-tomlinson** [comment]
@philbooth - updated with `[cC]hildView`.

File: app/scripts/views/app.js [hunk 3]
```diff
@@ -69,9 +69,9 @@ define([
         if (currentView) {
           if (currentView instanceof View) {
             // if the View to display is the same as the current view, then
-            // the user is navigating from a subview back to the parent view.
+            // the user is navigating from a childView back to the parent view.
             // No need to re-render, but notify interested parties of the event.
-            self._notifications.trigger('navigate-from-subview', options);
+            self._notifications.trigger('navigate-from-childView', options);
```
Comment (shane-tomlinson) [inline app/scripts/views/app.js]: We have `navigate-from-childView` but `show-child-view`. We should standardize here too. Maybe `show-childView`?

Comment (shane-tomlinson) [inline app/scripts/views/app.js]: `show-childView` mixes dashes and snake case.

Comment (shane-tomlinson) [inline app/scripts/views/app.js]: erm.. camel

Comment (philbooth) [inline app/scripts/views/app.js]: Hmmm, not sure how I feel about mixing camel and hypens. If something is hyphened it should be all hyphened, no?

Comment (shane-tomlinson) [inline app/scripts/views/app.js]: It's now all hyphenated

**Comment by shane-tomlinson** [comment]
From https://www.traviscistatus.com/:

> Identified - Heroku is experiencing platform issues which result in our API not being available on both travis-ci.org and travis-ci.com. See https://status.heroku.com/incidents/820

> Heroku - Major Outage

**Comment by philbooth** [comment]
Pending resolution of the CI problem, this looks good to me.

**Comment by shane-tomlinson** [comment]
Boo travis.

**Pull request merged**